### PR TITLE
feat(cra): move to react-docgen-typescript-plugin

### PR DESCRIPTION
**Pull request by mrmckeb**
This updates the CRA preset to mirror the config as of #11106.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -58,7 +58,7 @@
     "husky": "^4.2.3",
     "jest": "24.9.0",
     "lint-staged": "^10.0.8",
-    "prettier": "^1.19.1",
+    "prettier": "^2.0.5",
```
Comment (mrmckeb) [inline package.json:61]: Required for use with `import type...`.

File: packages/preset-create-react-app/src/index.ts [hunk 2]
```diff
@@ -118,21 +119,19 @@ export const webpack = (
     },
   );
 
-  // A temporary fix to align with Storybook 6.
-  const { typescriptOptions } = options;
-  const tsDocgenRule =
-    !typescriptOptions ||
+  // NOTE: These are set by default in Storybook 6.
+  const {
+    typescriptOptions = {
```
Comment (shilman) [inline packages/preset-create-react-app/src/index.ts:124]: Can we somehow keep this in sync with Storybook defaults? If I understand this code, adding adefault for `reactDocgen` "fixes the bug", but the `reactDocgenTypescriptOptions` are still out of sync with Storybook's:

https://github.com/storybookjs/storybook/blob/next/lib/core/src/server/config/defaults.js#L4-L7

Comment (mrmckeb) [reply packages/preset-create-react-app/src/index.ts:124]: These actually should already come through in Storybook v6 - so this is only for older Storybook versions. This default above would only occur if we didn't receive the defaults from Storybook.

File: packages/preset-create-react-app/src/index.ts [hunk 3]
```diff
@@ -118,21 +119,19 @@ export const webpack = (
...
+      reactDocgen: 'react-docgen-typescript',
+      reactDocgenTypescriptOptions: {},
```
Comment (shilman) [inline packages/preset-create-react-app/src/index.ts:126]: don't we need to add the extra options here?

**Pull request merged**